server/stream.go

```diff
@@ -4189,6 +4198,12 @@ func getExpectedLastSeqPerSubjectForSubject(hdr []byte) string {
 	return string(getHeader(JSExpectedLastSubjSeqSubj, hdr))
 }
 
+// Fast lookup of the message TTL.
+func getMessageTTL(hdr []byte) int64 {
```

Comment:
I think for this since it is a header it should be a parseable duration vs an integer.

E.g. `30s, 4h, 7d`

This will resolve the nanos issue.

We should reject if < 1s. So no `20ms` etc.

Also we should reject messages with per ttl set and the stream is not configured to opt-in. I don't think we need to do this at the lower store level but here we should.

Reply:
Updated so that it parses as a duration if it can, otherwise it reads the value as whole seconds. So `30s`, `1m` etc work, as does just `30` for 30 seconds.